packages/flutter_test/test/controller_test.dart

```diff
@@ -0,0 +1,277 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/gestures.dart';
+
+void main() {
+  testWidgets(
+    'WidgetTester.drag must break the offset into multiple parallel components if'
+    'the drag goes outside the touch slop values',
+    (WidgetTester tester) async {
+      // The first Offset in every sub array (ie. offsetResults[i][0]) is (touchSlopX, touchSlopY).
+      // The second Offset ... (ie. offsetResults[i][1]) will be the total move offset.
+      // The remaining values ... are the expected separated drag offsets.
+      final List<List<Offset>> offsetResults = <List<Offset>>[
```

Comment:
trivial nit: for consistency, the literals that look like int literals should have trailing `.0`s.